src/ws/service/friendships_service.rs

```diff
@@ -158,16 +160,41 @@ impl FriendshipsServiceServer<SocialContext> for MyFriendshipsService {
         }
     }
 
-    #[tracing::instrument(
-        name = "RPC SERVER > Update Friendship Event",
-        skip(_request, _context)
-    )]
+    #[tracing::instrument(name = "RPC SERVER > Update Friendship Event", skip(request, context))]
     async fn update_friendship_event(
         &self,
-        _request: UpdateFriendshipPayload,
-        _context: Arc<SocialContext>,
+        request: UpdateFriendshipPayload,
+        context: Arc<SocialContext>,
     ) -> UpdateFriendshipResponse {
-        todo!()
+        // Get user id with the given Authentication Token.
+        let user_id = get_user_id_from_request(
+            &request.clone().auth_token.unwrap(),
+            context.synapse.clone(),
+            context.users_cache.clone(),
+        )
+        .await;
+
+        // Process room event
```

Comment:
Can we explain here what processing a room means and why we need it for updating a friendship?